core/src/main/java/com/uber/m3/tally/ScopeBuilder.java

```diff
@@ -55,6 +57,7 @@
     protected String separator = DEFAULT_SEPARATOR;
     protected ImmutableMap<String, String> tags;
     protected Buckets defaultBuckets = DEFAULT_SCOPE_BUCKETS;
+    protected ScopeSanitizer sanitizer = new ScopeSanitizerBuilder().build();
```

Comment:
Let's make this opt-in -- by default sanitizer should be no-op

Reply:
Do you mean having it as `null` by default in the builder? 

This would be a little bit tricky. Then the ScopeImpl will have to do this null check everything, is that okay? (I would be easier if this is Kotlin and we can use `?` operator, sigh)

Reply:
NVM, looks like I will apply another comment from you and Andrey in https://github.com/uber-java/tally/pull/97#discussion_r618596672